File: JobRecruitment/src/main/webapp/src/app/model/Job.ts
import {BaseEntity} from "./BaseEntity";
import {Contractor} from "./Contractor";

export class Job extends BaseEntity{
  title : string;
  description : string;
  jobType : string;
  remote: boolean;
  minExperience: string;
  minCompensation: number;
  devType: string;
  techs: string;
  availablePos: number;
  dateAdded: Date;
  contractor: Contractor;


  constructor(id: number, title: string, description: string, jobType: string, remote: boolean, minExperience: string, minCompensation: number, devType: string, techs: string, availablePos: number, dateAdded: Date, contractor: Contractor) {
    super(id);
    this.title = title;
    this.description = description;
    this.jobType = jobType;
    this.remote = remote;
    this.minExperience = minExperience;
    this.minCompensation = minCompensation;
    this.devType = devType;
    this.techs = techs;
    this.availablePos = availablePos;
    this.dateAdded = dateAdded;
    this.contractor = contractor;
  }
}

File: JobRecommender/model/Job.py
class Job:
    def __init__(self, id, title, description, jobType, remote, minExperience, minCompensation, devType, techs,
                 availablePos, dateAdded, contractor):
        self.id = id
        self.title = title
        self.description = description
        self.jobType = jobType
        self.remote = remote
        self.minExperience = minExperience
        self.minCompensation = minCompensation
        self.devType = devType
        self.techs = techs
        self.availablePos = availablePos
        self.dateAdded = dateAdded
        self.contractor = contractor

    def serialize(self):
        return {
            'id': self.id,
            'title': self.title,
            'description': self.description,
            'jobType': self.jobType,
            'remote': self.remote,
            'minExperience': self.minExperience,
            'minCompensation': self.minCompensation,
            'devType': self.devType,
            'techs': self.techs,
            'availablePos': self.availablePos,
            'dateAdded': self.dateAdded,
            'contractor': None
        }

    def __str__(self):
        return str(self.id) + "|" + str(self.title) + "|" + str(self.description) + "|" + str(self.jobType) + "|" + str(
            self.remote) + "|" + str(self.minExperience) + "|" + \
               str(self.minCompensation) + "|" + str(self.devType) + "|" + str(self.techs) + "|" + str(
            self.availablePos)
